crates/dids/src/document.rs

```diff
@@ -10,8 +10,8 @@ pub struct Document {
     pub controller: Option<Vec<String>>,
     #[serde(rename = "alsoKnownAs", skip_serializing_if = "Option::is_none")]
     pub also_known_as: Option<Vec<String>>,
-    #[serde(rename = "verificationMethod")]
-    pub verification_method: Vec<VerificationMethod>,
+    #[serde(rename = "verificationMethod", skip_serializing_if = "Option::is_none")]
+    pub verification_method: Option<Vec<VerificationMethod>>,
```

Comment:
How do you all feel about this change? 6a54d270387965d396529f1311174b2a1a22fdd8 has some reasoning -- did-core and the web5 spec disagree so I'm not sure who's right. This example DID document on https://tbd.website/.well-known/did.json doesn't have a `VerificationMethod` which is what sparked this change.

Maybe it's worth a separate PR for discussion aside from the `did:web` resolution PR?

Reply:
yes, though I am wondering whether this guidance should still apply to the rust code.
originally it was to support a consistent feature set across languages that handled optional/polyglot type properties consistenly
however, if rust is able to directly support the did core spec then maybe our guidance is superfluous

suffice to say - in this case I would be fine to follow DID core and ignore our spec - but open to counter points

Reply:
Okay cool this is a great matter to work through. I'm in favor of enabling support for the DID Core spec over our `web5-spec` here in Rust. However, I propose we don't implement that change here, because there are more implications which need attending to. Primarily, the ones which come to mind are:

1. did:jwk requires exactly one verification method, and so we should add error cases
2. did:dht requires at least one verification method, and so we should add error cases
3. we'll need to also account for this change in the [APID](https://github.com/TBD54566975/web5-rs/pull/229) and consider the downstream implications for Kotlin & Swift target languages

@enmand could we parse these changes out of this PR, and then I stubbed in #231 for subsequent work

Reply:
Sounds good! I've opened https://github.com/TBD54566975/web5-rs/pull/233 with the changes to enable optional verification methods and rebased it out of this PR.